src/test/Makefile

```diff
@@ -517,12 +517,9 @@ GTEST_DIR = ../../lib/test/gtest
 
 CC  := clang-18
 CXX := clang++-18
-ifeq ($(shell which $(CC) 2>/dev/null),)
+ifeq ($(shell which $(CC) 2>/dev/null && which $(CXX) 2>/dev/null),)
```

Comment:
this doesn't work out the way we would like, It doesn't catch clang++-18 not being present. I got it to work this way. 
```makefile
CC  := clang-18
CXX := clang++-18
ifeq ($(and $(shell which $(CC) 2>/dev/null), $(shell which $(CXX) 2>/dev/null)),)
$(info Falling back to 'clang')
CC  := clang
CXX := clang++
endif
```
I tested that it uses clang-18 and clang++-18 when both are present by doing `sudo ln -s /usr/bin/clang++ /usr/local/bin/clang++-18` 
example output
```console
running test_baro_ms5611_unittest: PASS
compiling ../main/blackbox/blackbox_encoding.c 
mkdir -p ../../obj/test/blackbox_encoding_unittest/blackbox/
clang-18 -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/blackbox/blackbox_encoding.c -o ../../obj/test/blackbox_encoding_unittest/blackbox/blackbox_encoding.c.o
compiling ../main/common/encoding.c 
mkdir -p ../../obj/test/blackbox_encoding_unittest/common/
clang-18 -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/common/encoding.c -o ../../obj/test/blackbox_encoding_unittest/common/encoding.c.o
compiling ../main/common/printf.c 
mkdir -p ../../obj/test/blackbox_encoding_unittest/common/
clang-18 -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/common/printf.c -o ../../obj/test/blackbox_encoding_unittest/common/printf.c.o
compiling ../main/common/typeconversion.c 
mkdir -p ../../obj/test/blackbox_encoding_unittest/common/
clang-18 -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/common/typeconversion.c -o ../../obj/test/blackbox_encoding_unittest/common/typeconversion.c.o
compiling unit/blackbox_encoding_unittest.cc 
mkdir -p ../../obj/test/blackbox_encoding_unittest/
clang++-18 -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu++14 --coverage -Iunit -I../main  -c unit/blackbox_encoding_unittest.cc -o ../../obj/test/blackbox_encoding_unittest/blackbox_encoding_unittest.o
```
I then removed the symlink and tested it falls back to both it works
```console
running test_baro_ms5611_unittest: PASS
compiling ../main/blackbox/blackbox_encoding.c 
mkdir -p ../../obj/test/blackbox_encoding_unittest/blackbox/
clang -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/blackbox/blackbox_encodin
g.c -o ../../obj/test/blackbox_encoding_unittest/blackbox/blackbox_encoding.c.o
compiling ../main/common/encoding.c                      
mkdir -p ../../obj/test/blackbox_encoding_unittest/common/
clang -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/common/encoding.c -o ../.
./obj/test/blackbox_encoding_unittest/common/encoding.c.o                          
compiling ../main/common/printf.c                 
mkdir -p ../../obj/test/blackbox_encoding_unittest/common/                                                                                                                                                                             clang -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/common/printf.c -o ../../
obj/test/blackbox_encoding_unittest/common/printf.c.o                                                              
compiling ../main/common/typeconversion.c 
mkdir -p ../../obj/test/blackbox_encoding_unittest/common/
clang -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu99 --coverage -D_GNU_SOURCE -Iunit -I../main  -c ../main/common/typeconversion.c -
o ../../obj/test/blackbox_encoding_unittest/common/typeconversion.c.o
compiling unit/blackbox_encoding_unittest.cc 
mkdir -p ../../obj/test/blackbox_encoding_unittest/
clang++ -g -ggdb3 -Wall -Wextra -O0 -DUNIT_TEST -isystem ../../lib/test/gtest/inc -MMD -MP -pipe -Wno-c99-designator -Werror -fblocks -pthread -std=gnu++14 --coverage -Iunit -I../main  -c unit/blackbox_encoding_unittest.cc -o ../..
/obj/test/blackbox_encoding_unittest/blackbox_encoding_unittest.o
```